mars-sim-core/src/main/java/com/mars_sim/core/map/IntegerMapData.java

```diff
@@ -66,10 +67,8 @@ public class IntegerMapData implements MapData {
 	private int pixelHeight;
 	/* The cache for the last rho */
 	private double rhoCache;
-	/* The cache for the last phi. */
-	private double centerPhiCache;
-	/* The cache for the last theta. */
-	private double centerThetaCache;
+	/* The cache for the last center. */
+	private Coordinates centerCache;
```

Comment:
It's more efficient to simply pass 2 primitive data members namely, phi and theta) than the Coordinates object.

Both the regular cpu0() or the openCL's gpu() methods needs to processs the raw phi and theta.

Inside cpu0(), it calls for convertRectIntToSpherical(). Inside it needs the value of phi and theta.

I made this change recently to speed up the mineral layer rendering. 

Getting rid of using coordinates help with the efficiency tremendously.